config/routes.rb

```diff
@@ -17,6 +17,8 @@
   end
 
   resources :nodes, only: [:index, :show, :destroy] do
+    post :assign
```

Comment:
Right now the selection the user performs is master selection. The rest of the nodes will be assigned minion/worker role. That's why I proposed to use `assign_master` (maybe mapped to `assign-master` route). In this action we could set the master role to the selected node, set the minion/worker to the rest of the nodes, and that's it. Depending on how the setup on the UI is we would call the orchestration at the end (after setting roles for all machines involved in the cluster), or don't call the orchestration at all (if that's another `Bootstrap now` step in the UI, that I don't know right now).